docs/docs/pages/guides/multistep-deposit.mdx

```diff
@@ -0,0 +1,190 @@
+---
+title: Multi-Step Deposits
+description: How to configure an order for more complex interactions.
+---
+
+import { Callout } from 'vocs/components'
+
+# Multi-Step Deposits
+
+The Omni SDK allows you to compose multiple contract interactions on the destination chain within a single cross-chain order. This powerful feature enables complex workflows, such as depositing funds into one protocol and then staking the resulting token in another, all atomically from the user's perspective.
+
+This guide demonstrates how to configure a multi-step order using the `useOrder` hook and the `calls` parameter.
+
+## Example Scenario
+
+Imagine a user wants to deposit USDC on a source chain (e.g., Ethereum) and have it automatically deposited into a stablecoin vault on a destination chain (e.g., Optimism). The vault issues a receipt token (let's call it `newUSD`). The user then wants to stake this `newUSD` in a separate staking contract on the same destination chain, receiving a final staked token (let's call it `stkNewUSD`).
+
+The required steps on the destination chain are:
+
+1.  Deposit USDC into the Vault contract (receiving `newUSD`).
+2.  Approve the Staking contract to spend the received `newUSD` tokens.
+3.  Stake the `newUSD` tokens in the Staking contract for the user (receiving `stkNewUSD`).
+
+We can bundle these destination chain actions into a single Omni SDK order.
+
+## Key Considerations
+
+*   **Token Approvals via SolverNet Middleman:** When executing `calls`, the solver acts through a proxy contract called the `SolverNet Middleman` on the destination chain. This Middleman contract is the actual `msg.sender` for your target contract calls. In our example, the Middleman receives the `newUSD` from the vault. Consequently, the *Middleman* must be the one to call `approve` on the `newUSD` token contract, granting the Staking contract permission to spend its tokens.
```

Comment:
This bit isn't accurate. The calls are only proxied through the middleman if the actual call is made to the middleman.

The "silent" proxy (the contract that executes every call) is the SolverNetExector.